src/Models/ClosureTable.php

```diff
@@ -128,32 +115,10 @@ public function moveNodeTo($ancestorId = null)
      */
     protected function unbindRelationships()
     {
-        $table = $this->getPrefixedTable();
         $ancestorColumn = $this->getAncestorColumn();
         $descendantColumn = $this->getDescendantColumn();
 
-        $query = "
-            DELETE FROM {$table}
-            WHERE {$descendantColumn} IN (
-              SELECT d FROM (
-                SELECT {$descendantColumn} AS d FROM {$table}
-                WHERE {$ancestorColumn} = ?
-              ) AS dct
-            )
-            AND {$ancestorColumn} IN (
-              SELECT a FROM (
-                SELECT {$ancestorColumn} AS a FROM {$table}
-                WHERE {$descendantColumn} = ?
-                AND {$ancestorColumn} <> ?
-              ) AS ct
-            )
-        ";
-
-        $this->getConnection()->delete($query, [
-            $this->descendant,
-            $this->descendant,
-            $this->descendant
-        ]);
+        static::whereIn($descendantColumn, static::select([$descendantColumn])->where($ancestorColumn, $this->descendant)->get())->whereIn($ancestorColumn, static::select([$ancestorColumn])->where($descendantColumn, $this->descendant)->where($ancestorColumn, '<>', $this->descendant)->get())->delete();
```

Comment:
1. This is completely unreadable compared to the native query.
2. This way you make 3 queries against the database instead of one and, by using `static::select(...)->get()`, load data into memory prior to using it. It's a performance hit in case there are quite a few nodes.